src/main/java/hudson/plugins/sshslaves/SSHConnector.java

```diff
@@ -259,8 +261,9 @@ public SSHConnector(int port, StandardUsernameCredentials credentials, String us
         this.prefixStartSlaveCmd = fixEmpty(prefixStartSlaveCmd);
         this.suffixStartSlaveCmd = fixEmpty(suffixStartSlaveCmd);
         this.launchTimeoutSeconds = launchTimeoutSeconds == null || launchTimeoutSeconds <= 0 ? null : launchTimeoutSeconds;
-        this.maxNumRetries = maxNumRetries == null || maxNumRetries <= 0 ? 0 : maxNumRetries;
-        this.retryWaitTime = retryWaitTime == null || retryWaitTime <= 0 ? 0 : retryWaitTime;
+        final Integer zero = 0;
+        this.maxNumRetries = maxNumRetries == null || maxNumRetries <= 0 ? zero : maxNumRetries;
+        this.retryWaitTime = retryWaitTime == null || retryWaitTime <= 0 ? zero : retryWaitTime;
```

Comment:
@stephenc 
I'll change the order of conditions to <code>this.maxNumRetries = maxNumRetries != null && maxNumRetries > 0 ? maxNumRetries : 0;</code>. It will prevent a pointless boxing-unboxing cycle, because Integer will be taken as a return type of the operator